src/main/java/athena/timetable/Timetable.java

```diff
@@ -85,18 +135,229 @@ private void populateTimetable() {
+     * @param startHour The starting hour in 24-hour representation.
+     * @param endHour   The ending hour in 24-hour representation.
+     * @return A string that represents the bottom border for a day in the timetable.
+     */
+    private String drawBottomBorder(int startHour, int endHour) {
+        String row = "";
+        row += BOX_CORNER + DAY_BOX_HORIZONTAL_BORDER + BOX_CORNER;
+        for (int i = startHour; i < endHour; i++) {
+            row += TASK_BOX_HORIZONTAL_BORDER + BOX_CORNER;
+        }
+        row += "\n";
+        return row;
+    }
+
+    /**
+     * Utility method to fit a string into a given length.
+     * If the string is shorter than desired, pad it with spaces.
+     * If the string is longer than desired, truncate it and replace the end with "..".
+     *
+     * @param string The string to fit.
+     * @param maxLen The desired length.
+     * @return The string that is modified to fit into a given length.
+     */
+    private String shortenOrPadString(String string, int maxLen) {
+        if (string.length() == maxLen) {
+            return string;
+        } else if (string.length() < maxLen) {
+            return String.format("%" + -maxLen + "s", string);
+        }
+        return String.format("%s..", string.substring(0, maxLen - 2));
+    }
+
+    /**
+     * Finds a task in a task list that starts at the given hour.
+     *
+     * @param taskList Task list to search through.
+     * @param hour     Starting time of the task.
+     * @return The task that starts at the given hour. Null if doesn't exist.
+     */
+    private Task findTaskAtHour(TaskList taskList, int hour) {
+        for (Task task : taskList.getTasks()) {
+            int taskHour = task.getTimeInfo().getStartTime().getHour();
+            if (taskHour == hour) {
+                return task;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Generates a row for a day in the timetable containing the task information desired.
+     *
+     * @param day            The TimetableDay object for this day.
+     * @param startHour      The starting hour in 24-hour representation.
+     * @param endHour        The ending hour in 24-hour representation.
+     * @param taskInfoWriter A Function that extracts the information desired from a task.
+     * @return A string containing the task information desired.
+     */
+    private String drawTimetableDayRow(TimetableDay day, int startHour, int endHour,
+                                       Function<Task, String> taskInfoWriter) {
+        String row = "";
+
+        for (int hour = startHour; hour < endHour; hour++) {
+            Task task = findTaskAtHour(day.getTaskList(), hour);
+            if (task == null) {
+                row += EMPTY_TASK_BOX;
+                continue;
+            }
+
+            int duration = task.getTimeInfo().getDuration();
+            // TODO: better handle tasks exceeding sleep time, currently it just cuts off at sleep time
```

Comment:
we can just throw an error that says that the tasknumber is conflict with another, I'll handle this in the allocator PR

Reply:
Ok will leave this to you